### PR TITLE
Port Instructions to meta crate : part 2, typevar and operands

### PR DESCRIPTION
**Pull request by bnjbvr**
Based off https://github.com/CraneStation/cranelift/pull/705

### CODE REVIEW

File: cranelift-codegen/meta/src/cdsl/typevar.rs [hunk 1]
```diff
@@ -0,0 +1,743 @@
...
+        if self.lanes.contains(&1) {
+            copy.bools.insert(1);
+        }
+        copy
+    }
+
+    /// Return a TypeSet describing the image of self across halfwidth.
+    fn half_width(&self) -> TypeSet {
+        let mut copy = self.clone();
+        copy.ints = NumSet::from_iter(self.ints.iter().map(|&x| if x > 8 { x / 2 } else { x }));
```
Comment (jgouly) [inline cranelift-codegen/meta/src/cdsl/typevar.rs]: This seems slightly different from the Python, but maybe it doesn't make a difference in practise.

    [x//2 for x in self.ints if x > 8]
 
This filters out elements less than 8, but the rust code keeps them as-is?

**Comment by jgouly** [comment]
Looked just at the Typevar stuff, it's pretty dense. Looks good, pretty much a straight port from Python.

File: cranelift-codegen/meta/src/cdsl/typevar.rs (hunk 1 above)
Comment (bnjbvr) [reply cranelift-codegen/meta/src/cdsl/typevar.rs]: Good catch, I probably need to filter before the map (the cases where this was applied probably had both 8 and 16 in the int set, hiding this issue).

**Comment by bnjbvr** [comment]
Thanks for the reviews!

**Pull request merged**